test_regress/driver.py

```diff
@@ -65,6 +65,145 @@ def __get__(self, owner_self, owner_cls):
         return self.fget()
 
 
+#######################################################################
+#######################################################################
+# SAIF parser utilities
+
+
+class SAIFSignalBit:
+    last_val: int
+    high_time: int
+    low_time: int
+    transitions: int
+
+    def __init__(self):
+        self.last_val = 0
+        self.high_time = 0
+        self.low_time = 0
+        self.transitions = 0
+
+    def aggregate(self, dt: int, new_val: int):
+        if new_val != self.last_val:
+            self.transitions += 1
+
```

Comment:
Please squash the blank lines inside functions - alternatively you can remove all blanks inserted, then "make format" will insert what's needed style-wise.